syntax = "proto3";
// namespace arkanoid::io // todo

enum ElementType {
    BALL = 0;
    PADDLE = 1;
    BRICK = 2;
}

message NetBall {
    float velocity_x = 1;
    float velocity_y = 2;
}

message NetPaddle {
}

message NetBrick {
    int32 duration = 1;
}

message ElementPosition {
    float x = 1;
    float y = 2;
}

message GameElement {
    int32 id = 1;
    ElementType type = 2;
    ElementPosition element_position = 3; 

    oneof specific { NetBall ball = 5; NetPaddle paddle = 6; NetBrick brick = 7;}; // todo
}

message GameUpdate {
    repeated GameElement element = 1;
}